### PR TITLE
nmcli - add requirements for arch linux

### PR DESCRIPTION
**Pull request by senorsmile**
##### SUMMARY

Add list of packages needed to be installed for Arch Linux. 

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

nmcli

##### ADDITIONAL INFORMATION

Before installing these packages, Ansible will complain about various errors about imports, depending on which packages are missing.

### CODE REVIEW

File: plugins/modules/net_tools/nmcli.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
       python-dbus (or python3-dbus, depending on the Python version in use), libnm-glib-dev.'
     - 'On openSUSE, the requirements can be met by installing the following packages: NetworkManager, python2-dbus-python (or
       python3-dbus-python), typelib-1_0-NMClient-1_0 and typelib-1_0-NetworkManager-1_0.'
+    - 'On Arch Linux, the requirements can be met by installing the following packages: networkmanager, libnm, libnm-glib, python-dbus, python-gobject'
```
Comment (aminvakil) [inline plugins/modules/net_tools/nmcli.py:33]:
```suggestion
    - 'On Arch Linux, the requirements can be met by installing the following packages: networkmanager, libnm, python-dbus, python-gobject'
```

**Review by aminvakil — COMMENTED** [review]
Is `libnm-glib` necessary too? From what I can see it is a legacy version of `libnm` and `libnm` is already included in your PR.
Also I don't have `libnm-glib` installed, but I could use `community.general.nmcli` module on my Arch (although it may be because of my simple tasks which are simple).
https://www.archlinux.org/packages/extra/x86_64/libnm/
https://www.archlinux.org/packages/community/x86_64/libnm-glib/

**Comment by felixfontein** [comment]
It might make sense to wait with this PR until after #1113 is merged, because that changes the requirements a lot.

**Comment by senorsmile** [comment]
libnm-glib was the final package that I had to install to finally get it to work. 
I got specific glib errrors.  Tracing through the code, glib seemed to be the culprit.  Found the package that provided that, and finally got it working.

**Comment by mator** [comment]
lets see what is needed ... this is on the latest git version of C.G. (i.e. after #1113 got merged) :
```
# pacman -Q libnm-glib
error: package 'libnm-glib' was not found

# pacman -S networkmanager
resolving dependencies...
looking for conflicting packages...

Packages (11) bluez-libs-5.55-1  libmm-glib-1.14.6-1  libndp-1.7-2  libnewt-0.52.21-3  libnm-1.26.4-1  libpgm-5.3.128-1  libteam-1.31-1  mobile-broadband-provider-info-20190618-2  wpa_supplicant-2:2.9-7
              zeromq-4.3.3-2  networkmanager-1.26.4-1

Total Download Size:    7.41 MiB
Total Installed Size:  44.17 MiB

:: Proceed with installation? [Y/n] y
```
so `libnm` is already pulled by dependencies with `networkmanager` itself... 

install and start: 
```
[root@archlinux ansible.dev]# systemctl start NetworkManager
```
lets test a simple playbook: 
```

[mator@archlinux ansible.dev]$ cat playbooks/nmcli.yml
---
- name: community.general PR 1116
  hosts: localhost
  become: true
  gather_facts: false
  tasks:
  - name: 'Use the nmcli module with default parameters'
    nmcli:
      conn_name: 'testcon1'
      type: 'generic'
      state: 'present'

[mator@archlinux ansible.dev]$ ansible-playbook playbooks/nmcli.yml   
PLAY [community.general PR 1116] *********************************************************************************************************************************************************************************
TASK [Use the nmcli module with default parameters] **************************************************************************************************************************************************************
changed: [localhost]
PLAY RECAP *******************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```
testing results:
```
[mator@archlinux ansible.dev]$ nmcli  c sh
NAME                UUID                                  TYPE      DEVICE  
enp1s0              68f320c0-d3c4-4818-92e3-c057205fa4fc  ethernet  enp1s0  
docker0             c113c63f-7a7c-4712-8116-ccae3b87f2ae  bridge    docker0 
testcon1            59859eee-a6d6-491a-8da5-85ef11494b64  generic   --      
Wired connection 1  087b9193-cc80-3669-9618-cfae8932a04a  ethernet  --
```

so this PR is currently irrelevant/outdated..

Thanks.

**Comment by aminvakil** [comment]
@mator Could you please check if `python-dbus` and `python-gobject` are installed on your system or not?

I couldn't make the module work without these two.

I also think there is no harm to specify `libnm` specifically as this is mandatory.

**Comment by mator** [comment]
> Could you please check if `python-dbus` and `python-gobject` are installed on your system or not?

@aminvakil 

both packages are indeed installed : 
```
$ pacman -Q python-dbus python-gobject    
python-dbus 1.2.16-1
python-gobject 3.38.0-1
```

**Comment by aminvakil** [comment]
@mator So do you agree that `networkmanager`, `libnm`, `python-dbus` and `python-gobject` are necessary for this module to work on Arch Linux? Module worked for me without `libnm-glib` too.

> libnm-glib was the final package that I had to install to finally get it to work.
> I got specific glib errrors. Tracing through the code, glib seemed to be the culprit. Found the package that provided that, and finally got it working.

@senorsmile Could you please share the task that brought up glib errors? This PR also needs a rebase.

**Comment by aminvakil** [comment]
@senorsmile Could you please take another look at this?

**Comment by senorsmile** [comment]
Hi all.  Network Manager ended up having some specific issues that I couldn't get past, so I've switched back to Netctl.  
If I get a chance, I can try from scratch again to confirm.

**Comment by mator** [comment]
@senorsmile probably close this PR then... It is not practical to keep it opened without being able to review/rework. Thanks.

**Pull request closed without merging**